sunpy/net/scraper.py

```diff
@@ -286,13 +289,17 @@ def _httpfilelist(self, timerange):
                         log.debug(f"Converting retry_after failed: {e}")
                         retry_after = 2
                     log.debug(
-                        f"Got 429 while scraping {directory}, waiting for {retry_after} seconds before retrying."
+                        f"Got {http_err.code} while scraping {directory}, waiting for {retry_after} seconds before retrying."
                     )
                     sleep(retry_after)
-                    # Put this dir back on the queue
+                    if retry_counts.get(directory,0) > 4:
+                     log.debug(f"Exceeded maximum retry limit for {directory}")
+                     raise
+                    retry_counts[directory] = retry_counts.get(directory, 0) + 1
                     directories.insert(0, directory)
                     continue
-                raise
```

Comment:
Oh so this was part of the original code, this should be added back unless there was a reason to remove it?